snapshot/src/main/java/io/camunda/zeebe/snapshots/impl/FileBasedSnapshotStore.java

```diff
@@ -505,6 +508,10 @@ private void tryAtomicDirectoryMove(final Path directory, final Path destination
       LOGGER.warn("Atomic move not supported. Moving the snapshot files non-atomically.");
       Files.move(directory, destination);
     }
+
+    try (var stream = new FileOutputStream(destination.toFile())) {
+      stream.getFD().sync();
```

Comment:
Is opening a stream the only way to get a file descriptor? What about reusing the same as the file channel?

```java
try (final var channel = FileChannel.open(destination)) {
  channel.force(true);
}
```

I'm not sure if you can open a file channel to a directory though :sweat_smile:

Reply:
Reading through https://mail.openjdk.java.net/pipermail/nio-dev/2015-May/003140.html, it looks like it should be possible.